### DISCUSSION TITLE
How to change the script time limit in flash (non-flex) air application?

### DISCUSSION THREAD

**Discussion by ylazy** [airsdk/Adobe-Runtime-Support#3291] [Q&A]
Hi!

The default script time limit is 15 seconds. I can't find any instructions to change it. Using Adobe Animate I can change it via export settings. Now I'm using VSCode, are there any compiler options to use? I tried `acompc -help limit` but got `no keywords matched.`

Thanks!

**Comment by ajwfrost** [comment]
The compilers have a `-default-script-limits` flag:
https://airsdk.dev/docs/building/actionscript-compilers/application-compiler#editing-application-settings
First argument is max recursion depth, second is max script execution time.

Not entirely sure how you set that via VSCode/asconfig..?

**Reply by ajwfrost** [reply]
Ah - with asconfig, there's a `additionalOptions` setting that you can use for this I guess:
https://github.com/BowlerHatLLC/vscode-as3mxml/wiki/asconfig.json#additionaloptions

**Comment by ylazy** [comment]
```
mxmlc -default-size 240 240 \
      -default-frame-rate=24 \
      -default-script-limits 5000 10 -- \
      c:/myfiles/flex/misc/MainApp.mxml
```
I added `"-default-script-limits 1000 5"` to `additionalOptions` and the as3mxml extension showed me an error: `unknown configuragion variable default-script-limits 1000 5`
I changed it to `"-default-script-limits=1000,5"`, it seems to be correct as it didn't show errors anymore. But...

https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/errors/ScriptTimeoutError.html

I tried the example in the above link. On my machine it was hanging indefinitely:

```as
package {
    import flash.display.Sprite;
    import flash.errors.ScriptTimeoutError;

    public class ScriptTimeoutErrorExample extends Sprite {        
        private var keepLooping:Boolean = true;

        public function ScriptTimeoutErrorExample() {
            try {
                lockMachine();
            } 
            catch(e:ScriptTimeoutError) {
                trace(e);    // ScriptTimeoutError: Error #1502: A script has executed for longer than 15 seconds
                keepLooping = false;
            }
        }
        
        private function lockMachine():void {
            while(keepLooping){        
            }
        }
    }
}
```

btw the docs on airsdk.dev is empty:
https://airsdk.dev/reference/actionscript/3.0/flash/errors/ScriptTimeoutError.html

**Reply by ajwfrost** [reply]
A "while (1)" loop won't trigger a script timeout. There are only particular points within the ActionScript virtual machine that will trigger the check about long-running scripts - typically when entering/exiting functions I think. So although the script-stuck thread would have flagged it, the AS3 code would not be checking the flag if all it's doing is a tight loop...

Thanks for the spot on that error page, not sure what happened there! Will take a look..

thanks
